### PR TITLE
Add a simple sample for sorting

### PR DESCRIPTION
**Pull request by rubengees**
As @Rainer-Lang requested in #51 I have made a simple sample for sorting.
I am not quite sure if this is the best way to do it (See my comments) but it is something :)

### CODE REVIEW

File: app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+    /**
+     * Sorts the list ascending
+     */
+    @OnClick(R.id.button_ascending)
+    public void sortAscending() {
+        //Set the new comparator to the list
+        fastItemAdapter.getItemAdapter().withComparator(new AlphabetComparatorAscending());
+        //Reset the list to trigger the sorting
+        fastItemAdapter.setNewList(generateUnsortedList());
```
Comment (rubengees) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: Do we really need to reset the list here? Isn't there a method like `resort()` or something?
`notifyDataSetChanged()` did not help.

File: app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setAdapter(fastItemAdapter);
+
+        if (savedInstanceState != null) {
+            //noinspection unchecked
+            fastItemAdapter.getItemAdapter().withComparator((Comparator<SampleItem>)
+                    savedInstanceState.getSerializable("comparator"));
+        }
+
+        fastItemAdapter.setNewList(generateUnsortedList());
```
Comment (rubengees) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: Why does `sett()` not work, whereas `setNewList()` does work if there is already a Comparator from a previous instance state?

Comment (mikepenz) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: I can it add for `set()` to. Haven't done this as I thought it doesn't make so much sense but yeah

File: app/src/main/java/com/mikepenz/fastadapter/app/SampleActivity.java [hunk 3]
```diff
@@ -77,6 +77,7 @@ protected void onCreate(Bundle savedInstanceState) {
                         new PrimaryDrawerItem().withName(R.string.sample_radiobutton_item).withDescription(R.string.sample_radiobutton_item_descr).withSelectable(false).withIdentifier(11).withIcon(CommunityMaterial.Icon.cmd_radiobox_marked),
                         new PrimaryDrawerItem().withName(R.string.sample_swipe_list).withDescription(R.string.sample_swipe_list_descr).withSelectable(false).withIdentifier(12).withIcon(MaterialDesignIconic.Icon.gmi_format_align_left),
                         new PrimaryDrawerItem().withName(R.string.sample_endless_scroll_list).withDescription(R.string.sample_endless_scroll_list_descr).withSelectable(false).withIdentifier(13).withIcon(MaterialDesignIconic.Icon.gmi_long_arrow_down),
+                        new PrimaryDrawerItem().withName(R.string.sample_sort).withDescription(R.string.sample_sort_descr).withSelectable(false).withIdentifier(14).withIcon(MaterialDesignIconic.Icon.gmi_sort_by_alpha),
```
Comment (FabianTerhorst) [inline app/src/main/java/com/mikepenz/fastadapter/app/SampleActivity.java]: You should add a sample_sort_list_descr string like in the other examples.

File: app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java (hunk 1 above)
Comment (rubengees) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: Well at least I would expect that.-

File: app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+import butterknife.Bind;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+/**
+ * TODO: Describe Class
+ *
+ * @author Ruben Gees
+ */
+public class SortActivity extends AppCompatActivity {
```
Comment (FabianTerhorst) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: In the whole project is no class comment. I don´t know if this is needed here.

File: app/src/main/java/com/mikepenz/fastadapter/app/SampleActivity.java (hunk 3 above)
Comment (rubengees) [inline app/src/main/java/com/mikepenz/fastadapter/app/SampleActivity.java]: What do you mean? It is there.

Comment (FabianTerhorst) [inline app/src/main/java/com/mikepenz/fastadapter/app/SampleActivity.java]: The item strings from the other items are all named sample_name_*_list_descr*

File: app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java (hunk 4 above)
Comment (rubengees) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: Fixed.

File: app/src/main/res/layout/activity_sort.xml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="?attr/actionBarSize"
+        android:background="?attr/colorPrimary"
+        android:elevation="4dp" />
+
+    <RelativeLayout
+        android:id="@+id/frame_container"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_below="@id/toolbar">
```
Comment (rubengees) [inline app/src/main/res/layout/activity_sort.xml]: Ah we can't use an `AppBarLayout` as the design support lib is not added as a dependency. 
I wouldn't add it for this only though.

File: app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        getSupportActionBar().setHomeButtonEnabled(false);
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        //add the values which need to be saved from the adapter to the bundle
+        outState = fastItemAdapter.saveInstanceState(outState);
+        //We need to persist our comparator between orientation changes
+        outState.putSerializable("comparator", (Serializable)
+                fastItemAdapter.getItemAdapter().mComparator);
```
Comment (rubengees) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: Yes you're right. I change it right away.

File: app/src/main/res/layout/activity_sort.xml (hunk 5 above)
Comment (mikepenz) [inline app/src/main/res/layout/activity_sort.xml]: @rubengees it should be part of the `sample` as the `library-extensions` comes with it

File: app/src/main/java/com/mikepenz/fastadapter/app/SampleActivity.java (hunk 3 above)
Comment (FabianTerhorst) [inline app/src/main/java/com/mikepenz/fastadapter/app/SampleActivity.java]: I think there should have the name name. You can rename them all to the same string prefix when you want.

File: app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java (hunk 6 above)
Comment (rubengees) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: Done. Have a look.

File: app/src/main/res/layout/activity_sort.xml (hunk 5 above)
Comment (Rainer-Lang) [inline app/src/main/res/layout/activity_sort.xml]: @rubengees Then you can use AppBarLayout :)

**Comment by Rainer-Lang** [comment]
@rubengees Thanks for this pr :+1:

File: app/src/main/res/layout/activity_sort.xml (hunk 5 above)
Comment (rubengees) [inline app/src/main/res/layout/activity_sort.xml]: Done.

File: app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java (hunk 6 above)
Comment (Rainer-Lang) [inline app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java]: Looks good for me.

**Comment by mikepenz** [comment]
Seems fine now :). Thank you very much for this PR

**Pull request merged**